### ISSUE TITLE
AttributeError: 'dict' object has no attribute 'labels_suffix'

### ISSUE THREAD

**Issue by YiLin-Zhou** [talmolab/dreem#89]
## Bug description
Hi, unfortunately I'm getting a new error `AttributeError: 'dict' object has no attribute 'labels_suffix'` after pulling the most recent changes & running `sleap-train` again:
(Also, FYI that OSError seems to be windows specific, and only shows up after 'we had to kill ffmpeg to stop it', doesn't seem to affect function otherwise so I've just been ignoring it).

## Terminal error logs
```
Error executing job with overrides: []
Traceback (most recent call last):
  File "C:\Users\ylzhou\dreem\dreem\training\train.py", line 62, in run
    test_dataset = train_cfg.get_dataset(mode="test")
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\io\config.py", line 242, in get_dataset
    label_files, vid_files = self.get_data_paths(dataset_params)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\io\config.py", line 185, in get_data_paths
    labels_suff = dir_cfg.labels_suffix
                  ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute 'labels_suffix'

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
[2024-10-17 11:09:49,376][imageio_ffmpeg][WARNING] - We had to kill ffmpeg to stop it.
Exception ignored in: <function SleapDataset.__del__ at 0x0000018D988131A0>
Traceback (most recent call last):
  File "C:\Users\ylzhou\dreem\dreem\datasets\sleap_dataset.py", line 374, in __del__
    reader.close()
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\imageio\core\format.py", line 350, in close
    self._close()
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\imageio\plugins\ffmpeg.py", line 370, in _close
    self._read_gen.close()
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\imageio_ffmpeg\_io.py", line 379, in read_frames
    time.sleep(0.01)
OSError: [WinError 6] The handle is invalid
[2024-10-17 11:09:49,383][imageio_ffmpeg][WARNING] - We had to kill ffmpeg to stop it.
...
[2024-10-17 11:09:49,465][imageio_ffmpeg][WARNING] - We had to kill ffmpeg to stop it.
Exception ignored in: <function SleapDataset.__del__ at 0x0000018D988131A0>
Traceback (most recent call last):
  File "C:\Users\ylzhou\dreem\dreem\datasets\sleap_dataset.py", line 374, in __del__
    reader.close()
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\imageio\core\format.py", line 350, in close
    self._close()
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\imageio\plugins\ffmpeg.py", line 370, in _close
    self._read_gen.close()
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\imageio_ffmpeg\_io.py", line 379, in read_frames
    time.sleep(0.01)
OSError: [WinError 6] The handle is invalid
[2024-10-17 11:09:49,474][imageio_ffmpeg][WARNING] - We had to kill ffmpeg to stop it.
...
[2024-10-17 11:09:49,529][imageio_ffmpeg][WARNING] - We had to kill ffmpeg to stop it.
Exception ignored in: <function Popen.__del__ at 0x0000018DF9535760>
Traceback (most recent call last):
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\subprocess.py", line 1130, in __del__
    self._internal_poll(_deadstate=_maxsize)
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\subprocess.py", line 1575, in _internal_poll
    if _WaitForSingleObject(self._handle, 0) == _WAIT_OBJECT_0:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [WinError 6] The handle is invalid
```

## Config file
```
model:
  ckpt_path: null
  encoder_cfg: 
    model_name: "resnet18"
    backend: "timm"
    in_chans: 3
  d_model: 1024
  nhead: 8
  num_encoder_layers: 1
  num_decoder_layers: 1
  dropout: 0.1
  activation: "relu"
  return_intermediate_dec: False
  norm: False
  num_layers_attn_head: 2
  dropout_attn_head: 0.1
  embedding_meta: 
    pos:
        mode: "learned"
        normalize: true
    temp:
        mode: "learned"
  return_embedding: False
  decoder_self_attn: False

loss:
  neg_unmatched: false
  epsilon: 1e-4
  asso_weight: 1.0

#currently assumes adam. TODO adapt logic for other optimizers like sgd
optimizer:
  name: "Adam"
  lr: 0.001
  betas: [0.9, 0.999]
  eps: 1e-8
  weight_decay: 0.01

#currently assumes reduce lr on plateau
scheduler:
  name: "ReduceLROnPlateau"
  mode: "min"
  factor: 0.5
  patience: 10
  threshold: 1e-4
  threshold_mode: "rel"

tracker:
  window_size: 10
  use_vis_feats: true
  overlap_thresh: 0.01
  mult_thresh: true
  decay_time: null
  iou: null
  max_center_dist: 0.1

runner:
  metrics:
      train: ['num_switches']
      val: ['num_switches']
      test: ['num_switches']
  persistent_tracking:
      train: false
      val: false
      test: true

dataset:
  train_dataset:
      slp_files: ["/Users/ylzhou/Documents/DREEM/larva_dreem_100examples_dataset/train/CC24.1_fr1-30.slp", ...]
      video_files: ["/Users/ylzhou/Documents/DREEM/larva_dreem_100examples_dataset/train/cc18.4_fr1-30.avi", ...]
      padding: 5
      crop_size: 105
      chunk: false
      anchors: ["gut1", "gut2"]
      handle_missing: "centroid"
      augmentations: 
          Rotate:
              limit: 10
              p: 0.2
          GaussianBlur:
              blur_limit: [3,7]
              sigma_limit: 0
              p: 0.2
          RandomBrightnessContrast:
              brightness_limit: [-0.2, 0.2]
              contrast_limit: [-0.2, 0.2]
              p: 0.2


  val_dataset:
      slp_files: ["/Users/ylzhou/Documents/DREEM/larva_dreem_100examples_dataset/val/CC21.2_fr1-30.slp", ...]
      video_files: ["/Users/ylzhou/Documents/DREEM/larva_dreem_100examples_dataset/val/CC21.2_fr1-30.avi", ...]
      padding: 5
      crop_size: 105
      chunk: false
      handle_missing: "centroid"

  test_dataset:
      dir:
            path: '/Users/ylzhou/Documents/DREEM/larva_dreem_100examples_dataset/test'
            labels_suffix: '.slp'
            vid_suffix: '.avi'
      slp_files: ["/Users/ylzhou/Documents/DREEM/larva_dreem_100examples_dataset/test/CC2.1.slp"]
      video_files: ["/Users/ylzhou/Documents/DREEM/larva_dreem_100examples_dataset/test/CC2.1.avi"]
      padding: 5
      crop_size: 105
      chunk: False
      handle_missing: "centroid"

dataloader:
  train_dataloader:
    shuffle: true
    num_workers: 0
  val_dataloader:
    shuffle: false
    num_workers: 0
  test_dataloader:
    shuffle: false
    num_workers: 0

logging:
  logger_type: null
  name: "100example_augmentation_00"
  entity: null
  job_type: "train"
  notes: "100 examplea w/ ugmentation job 00"
  dir: "./logs"
  group: "logging"
  save_dir: './logs'
  project: "GTR"
  log_model: "all"

early_stopping:
  monitor: "val_loss"
  min_delta: 0.1
  patience: 10
  mode: "min"
  check_finite: true
  stopping_threshold: 1e-8
  divergence_threshold: 30

checkpointing:
  monitor: ["val_loss","val_num_switches"]
  verbose: true
  save_last: true
  dirpath: null
  auto_insert_metric_name: true
  every_n_epochs: 2

trainer:
  check_val_every_n_epoch: 1
  enable_checkpointing: true
  gradient_clip_val: null
  limit_train_batches: 1.0
  limit_test_batches: 1.0
  limit_val_batches: 1.0
  log_every_n_steps: 1
  max_epochs: 26
  min_epochs: 8

view_batch:
  enable: False
  num_frames: 0
  no_train: False
```

**Comment by aaprasad** [comment]
Hi @YiLin-Zhou, thanks for pointing that out. It should be fixed as of #90. Could you try pulling and rerunning your code and see if there is any other issue?

**Comment by YiLin-Zhou** [comment]
Hi @aaprasad, thanks for your quick response! I pulled, re-ran my code, got this additional error:

```
None not one of ['CSVLogger', 'TensorBoardLogger', 'WandbLogger'] or set to None, skipping logging
Error executing job with overrides: []
Traceback (most recent call last):
  File "C:\Users\ylzhou\dreem\dreem\training\train.py", line 81, in run
    _ = callbacks.extend(train_cfg.get_checkpointing())
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\io\config.py", line 423, in get_checkpointing
    dirpath = f"./models/{logging_params.group}/{logging_params.name}"
                          ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute 'group'

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
[2024-10-17 17:36:24,513][imageio_ffmpeg][WARNING] - We had to kill ffmpeg to stop it.
Exception ignored in: <function SleapDataset.__del__ at 0x0000021DC8DE3240>
```

**Comment by aaprasad** [comment]
@YiLin-Zhou got it will fix that! in the mean time set the `checkpointing.dirpath` so that it doesnt try to use the logger params and that should resolve it. E.G in your config file:
```YAML
checkpointing:
  monitor: ["val_loss","val_num_switches"]
  verbose: true
  save_last: true
  dirpath: null #set this to something (can even be ".")
  auto_insert_metric_name: true
  every_n_epochs: 2
```

**Comment by YiLin-Zhou** [comment]
@aaprasad I tried changing the `dirpath` to `.` and `/Users/ylzhou/Documents/DREEM/larva_dreem_100examples_dataset`, but either one of the two will result in this type of error log:

```
Sanity Checking: |                                                                                      | 0/? [00:00<?, ?it/s]C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:424: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=19` in the `DataLoader` to improve performance.
C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:424: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=19` in the `DataLoader` to improve performance.
Epoch 0:   0%|                                                                                         | 0/80 [00:00<?, ?it/s][2024-10-18 08:10:18,320][dreem.models][ERROR] - Failed on frame tensor([0], device='cuda:0') of video tensor([14], device='cuda:0')
Traceback (most recent call last):
  File "C:\Users\ylzhou\dreem\dreem\models\gtr_runner.py", line 193, in _shared_eval_step
    logits = self(instances)
             ^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\models\gtr_runner.py", line 97, in forward
    asso_preds = self.model(ref_instances, query_instances)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\models\global_tracking_transformer.py", line 93, in forward
    self.extract_features(ref_instances)
  File "C:\Users\ylzhou\dreem\dreem\models\global_tracking_transformer.py", line 127, in extract_features
    features = self.visual_encoder(crops)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\models\visual_encoder.py", line 141, in forward
    feats = self.feature_extractor(
            ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\timm\models\resnet.py", line 635, in forward
    x = self.forward_features(x)
        ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\timm\models\resnet.py", line 614, in forward_features
    x = self.conv1(x)
        ^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\conv.py", line 458, in forward
    return self._conv_forward(input, self.weight, self.bias)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\torch\nn\modules\conv.py", line 454, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Given groups=1, weight of size [64, 3, 7, 7], expected input[839, 6, 115, 115] to have 3 channels, but got 6 channels instead
[2024-10-18 08:10:18,339][dreem.models][ERROR] - Given groups=1, weight of size [64, 3, 7, 7], expected input[839, 6, 115, 115] to have 3 channels, but got 6 channels instead
...
Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
[2024-10-18 08:10:18,714][imageio_ffmpeg][WARNING] - We had to kill ffmpeg to stop it.
...
```

**Comment by YiLin-Zhou** [comment]
Will do, thanks!

**Comment by aaprasad** [comment]
I'm gna close this issue now as it should be resolved as of #92. Let me know if there are any other issues
